### ISSUE TITLE
Are widgets in metakernel_python supposed to work?

### ISSUE THREAD

**Issue by freifrauvonbleifrei** [Calysto/metakernel#163]
Hi,

[this issue (#12)](https://github.com/Calysto/metakernel/issues/12) sounded like the basic widget functionality should be working with metakernel. However, the Python metakernel does not display widgets while the IPykernel does: 

![screenshot from 2018-03-22 13-52-01](https://user-images.githubusercontent.com/9729409/37771797-ff05b3b6-2dd8-11e8-8484-6aa3f59c61ed.png)

![screenshot from 2018-03-22 13-51-43](https://user-images.githubusercontent.com/9729409/37771807-0a167e34-2dd9-11e8-8ad9-9dad9d34a949.png)

Is this an issue with my system, or does it reflect the current state of developments?

**Comment by dsblank** [comment]
Something must have changed. I will look into it. Thanks for reporting!

**Comment by freifrauvonbleifrei** [comment]
I noticed that there are some errors in the firefox console for metakernel python, which are weird because I can access the files by the given URLs in the browser.

```
Source map error: request failed with status 404
Resource URL: http://localhost:8888/static/components/preact-compat/index.js?v=f865e990e65ad27e3a2601d8adb48db1
Source Map URL: preact-compat.min.js.map[Learn More]

Source map error: request failed with status 404
Resource URL: http://localhost:8888/static/components/preact/index.js?v=00a2fac73c670ce39ac53d26640eb542
Source Map URL: preact.min.js.map[Learn More]

Source map error: request failed with status 404
Resource URL: http://localhost:8888/static/components/proptypes/index.js?v=c40890eb04df9811fcc4d47e53a29604
Source Map URL: index.js.map[Learn More]
```

And then, there is a whole bunch of these errors here, but seems the files under `/home/takluyver/Code/` are actually not there.
```
Error while fetching an original source: request failed with status 404
Source URL: http://localhost:8888/home/takluyver/Code/notebook/notebook/static/components/bootstrap/less/normalize.less[Learn More]
```

None of these show up with IPython. Hope this helps :)

**Comment by dsblank** [comment]
Finally got some time to look into this. It appears that since ipywidgets and ipython have been separated a bit, the formatters are no longer registered with IPythonDisplayFormatter. I *think* we need to register the widgets with the formatter, and then we'll be ok (on the front end). Anyone have any ideas?

**Comment by joequant** [comment]
This has been fixed in 0.25.0.  Can close this.
